java/runtime/src/main/java/org/ray/runtime/AbstractRayRuntime.java

```diff
@@ -67,18 +68,18 @@ public AbstractRayRuntime(RayConfig rayConfig, FunctionManager functionManager)
   @Override
   public <T> RayObject<T> put(T obj) {
     ObjectId objectId = objectStore.put(obj);
-    return new RayObjectImpl<>(objectId);
+    return new RayObjectImpl<T>(objectId, (Class<T>)(obj == null ? Object.class : obj.getClass()));
```

Comment:
It works OK, because message pack will save `s` as `NIL` type when serializing. And unpacker ignores the expected type when deserializing `NIL`.
So, null is always null.